From error27@gmail.com  Thu Jan 14 16:25:57 2010
From: Dan Carpenter <error27@gmail.com>
Date: Mon, 28 Dec 2009 19:01:34 +0200
Subject: Staging: otus: off by one in usbdrvwext_siwessid()
To: Luis.Rodriguez@Atheros.com
Cc: otus-devel@lists.madwifi-project.org, devel@driverdev.osuosl.org, Greg Kroah-Hartman <gregkh@suse.de>
Message-ID: <20091228170134.GF17645@bicker>
Content-Disposition: inline


A 33 char ESSID is too long and it could cause a buffer overflow
a couple lines below when we put a NULL terminator on the end.

Signed-off-by: Dan Carpenter <error27@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/otus/ioctl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/otus/ioctl.c
+++ b/drivers/staging/otus/ioctl.c
@@ -930,7 +930,7 @@ int usbdrvwext_siwessid(struct net_devic
 		return -EINVAL;
 
 	if (essid->flags == 1) {
-		if (essid->length > (IW_ESSID_MAX_SIZE + 1))
+		if (essid->length > IW_ESSID_MAX_SIZE)
 			return -E2BIG;
 
 		if (copy_from_user(&EssidBuf, essid->pointer, essid->length))
